docs/build.rb

```diff
@@ -0,0 +1,1151 @@
+      title_collisions = (folder_titles & page_titles).tally
+      raise "Duplicate folder/page titles, will cause issues: #{title_collisions}" if title_collisions.any?
+
+      # Ensure there are no files being migrated to multiple places
+      page_paths = to_enum.reject { |page| page[:path] }.map { |page| page[:title] }
+      duplicate_page_paths = page_paths.tally.select { |_name, count| count > 1 }
+      raise "Duplicate paths, will cause issues: #{duplicate_page_paths}" if duplicate_page_paths.any?
+
+      # Ensure new file paths are only alphanumeric and hyphenated
+      new_paths = to_enum.map { |page| page[:new_path] }
+      invalid_new_paths = new_paths.select { |path| File.basename(path) !~ /^[a-zA-Z0-9_-]*\.md$/ }
+      raise "Only alphanumeric and hyphenated file names required: #{invalid_new_paths}" if invalid_new_paths.any?
+    end
+
+    def available_paths
+      Dir.glob("#{WIKI_PATH}/**/*{.md,.textile}", File::FNM_DOTMATCH)
+    end
+
+    def ignored_paths
+      [
+        '_Sidebar.md',
+        'dev/_Sidebar.md',
+      ]
+    end
+
+    def existing_docs
+      existing_docs = Dir.glob('docs/**/*', File::FNM_DOTMATCH)
+      existing_docs
+    end
+
+    def each(&block)
+      config.each do |parent|
+        recurse(with_metadata(parent), &block)
+      end
+    end
+
+    def all_file_paths
+      to_enum.map { |item| item[:path] }.to_a
+    end
+
+    protected
+
+    # depth first traversal
+    def recurse(parent_with_metadata, &block)
+      block.call(parent_with_metadata)
+      parent_with_metadata[:children].to_a.each do |child|
+        child_with_metadata = with_metadata(child, parents: parent_with_metadata[:parents] + [parent_with_metadata])
+        recurse(child_with_metadata, &block)
+      end
+    end
+
+    def with_metadata(child, parents: [])
+      child = child.clone
+
+      if child[:folder]
+        parent_folders = parents.map { |page| page[:folder] }
+        child[:new_path] = File.join(*parent_folders, child[:folder], 'index.md')
+      else
+        path = child[:path]
+        base_name = child[:new_base_name] || File.basename(path)
+
+        # title calculation
+        computed_title = File.basename(base_name, '.md').gsub('-', ' ')
+        if child[:title].is_a?(Proc)
+          child[:title] = child[:title].call(computed_title)
+        else
+          child[:title] ||= computed_title
+        end
+
+        parent_folders = parents.map { |page| page[:folder] }
+        child[:new_path] = File.join(*parent_folders, base_name.downcase)
+      end
+
+      child[:parents] = parents
+      child[:has_children] = true if child[:children].to_a.any?
+
+      child
+    end
+
+    def without_prefix(prefix)
+      proc { |value| value.gsub(/^#{prefix}/, '') }
+    end
+
+    attr_reader :config
+  end
+
+  # Extracts markdown links from https://github.com/rapid7/metasploit-framework/wiki into a Jekyll format
+  # Additionally corrects links to Github / Twitter accounts
```

Comment:
Currently this seems to only be GitHub links unless there is some Twitter code I'm missing somewhere?
```suggestion
  # Additionally corrects links to Github accounts
```

Reply:
Agree, I just wanted to avoid confusion that we were also adding links to people's Twitter accounts since we don't do that at the moment.

Fixed in 48ccb93b550442af412eb79d0f37afee6bd7267e